packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -293,54 +327,46 @@ export const BuyMembershipForm = ({
                         If your validator account is not in your signer wallet, paste the account address to the field
                         below:
                       </TextMedium>
-                      <RowInline>
-                        <InputComponent id="select-validatorAccount" inputSize="l">
-                          <SelectAccount
-                            id="select-validatorAccount"
-                            name="validatorAccountCandidate"
-                            filter={(account) => !!validatorAddresses?.includes(account.address)}
-                          />
-                        </InputComponent>
-                        <ButtonPrimary
-                          square
-                          size="large"
-                          onClick={addValidatorAccount}
-                          disabled={!isValidValidatorAccount}
-                          className="add-button"
-                        >
-                          <PlusIcon />
+                      {validatorAccountCandidates?.map((account, index) => (
+                        <Row>
+                          <RowInline>
+                            <InputComponent id="select-validatorAccount" inputSize="l">
+                              <SelectAccount
+                                id="select-validatorAccount"
+                                name={`validatorAccountCandidates[${index}]`}
+                                filter={(account) => validatorAccountSelectorFilter(index, account)}
+                              />
+                            </InputComponent>
+                            <ButtonGhost
+                              square
+                              size="large"
+                              onClick={() => {
+                                removeValidatorAccount(index)
+                              }}
+                            >
+                              <CrossIcon />
+                            </ButtonGhost>
+                          </RowInline>
+                          {!isValidValidatorAccount(account) && (
+                            <RowInline gap={2}>
+                              <TextSmall error>
+                                <InputNotificationIcon>
+                                  <AlertSymbol />
+                                </InputNotificationIcon>
+                              </TextSmall>
+                              <TextSmall error>
+                                This account is neither a validator controller account nor a validator stash account.
+                              </TextSmall>
+                            </RowInline>
+                          )}
+                        </Row>
+                      ))}
+                      <RowInline justify="end">
+                        <ButtonPrimary size="small" onClick={addValidatorAccount} className="add-button">
```

Comment:
Now this can be:
```suggestion
                        <ButtonPrimary
                           size="small"
                           className="add-button"
                           onClick={() => addValidatorAccount(validatorAccount.length, undefined)}
                           >
```